app/models/email_address.rb

```diff
@@ -38,6 +38,16 @@ def find_with_email(email)
       find_by(email_fingerprint: email_fingerprint)
     end
 
+    def update_last_sign_in_at_on_user_id_and_email(user_id: , email:)
+      return nil if !email.is_a?(String) || email.empty?
```

Comment:
I tend to prefer casting instead of type checks, what do you think of?

```suggestion
      return nil if email.to_s.empty?
```

Reply:
Yep, makes sense. I copied from the method above, so I'll change it there too.